jenkins/spark-premerge-build.sh

```diff
@@ -75,10 +75,13 @@ mvn_verify() {
           -DwildcardSuites=com.nvidia.spark.rapids.ConditionalsSuite,com.nvidia.spark.rapids.RegularExpressionSuite,com.nvidia.spark.rapids.RegularExpressionTranspilerSuite
     done
 
+    # Calculate parallelism based on GPU memory. When parallelism > 8, as it increases, the test speed will become slower and slower. So we set the maximum parallelism to 8.
+    GPU_MEM_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int((MAX - 2 * 1024) / ((1.5 * 1024) + 750))}'`
+    export TEST_PARALLEL=$(( $GPU_MEM_PARALLEL < 8 ? $GPU_MEM_PARALLEL : 8 ))
```

Comment:
I am confused. This is the same as what we already have in run_pyspark_from_build.sh
https://github.com/NVIDIA/spark-rapids/blob/2cf59346fd66e1d500a16e8107068dc9e20c3585/integration_tests/run_pyspark_from_build.sh#L113-L134

Except it is missing the maximum of 8. 

Could we please add an option to run_pyspark_from_build.sh so we can set a maximum test parallelism. I am okay with a maximum of 8 as the default. That way if we ever do need to change this code, or add in a new set of tests, then we just get it by default instead of trying to change it all in multiple different places.

Reply:
@revans2  Adding a maximum parallelism sounds good to me, updated!

Thanks a lot for the review!